### PR TITLE
Add command line interface for plotting

### PR DESCRIPTION
**Pull request by dweindl**

### CODE REVIEW

File: petab/visualize/cli.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+"""Command line interface for visualization"""
```
Comment (dilpath) [inline petab/visualize/cli.py:1]:
```suggestion
"""Command-line interface for visualization."""
```

File: petab/visualize/cli.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def _parse_cli_args():
+    """Parse command line arguments"""
+
+    parser = argparse.ArgumentParser(
+        description='Create PEtab visualizations.')
+
+    parser.add_argument('-y', '--yaml', dest='yaml_file_name', required=True,
+                        help='PEtab YAML problem filename')
```
Comment (dilpath) [inline petab/visualize/cli.py:16]:
```suggestion
                        help='PEtab problem YAML filename')
```

File: petab/visualize/cli.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                        required=False,
+                        help='PEtab simulation filename')
+    parser.add_argument('-o', '--output-directory', dest='output_directory',
+                        required=True, help='Output directory')
+    parser.add_argument('-v', '--visualizations', required=False,
+                        dest='visualization_file_name',
+                        help='PEtab visualization specification filename')
+    parser.add_argument('--style', required=False,
+                        dest='style_file_name',
+                        help='Matplotlib style file')
```
Comment (dilpath) [inline petab/visualize/cli.py:27]: This is nice, adding an example at some point to show how Seaborn defaults could be used could be helpful

File: petab/visualize/cli.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    return parser.parse_args()
+
+
+def _petab_visualize_main():
+    """Entrypoint for visualization command line interface"""
```
Comment (dilpath) [inline petab/visualize/cli.py:32]:
```suggestion
    """Entrypoint for visualization command-line interface."""
```

File: petab/visualize/cli.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+"""Command line interface for visualization"""
+import argparse
+from pathlib import Path
+import matplotlib.pyplot as plt
+from .plot_data_and_simulation import plot_problem
+from .. import Problem, get_simulation_df, get_visualization_df
+
+
+def _parse_cli_args():
+    """Parse command line arguments"""
```
Comment (dilpath) [inline petab/visualize/cli.py:10]:
```suggestion
    """Parse command-line arguments."""
```

File: petab/visualize/cli.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+def _petab_visualize_main():
+    """Entrypoint for visualization command line interface"""
+    args = _parse_cli_args()
+
+    petab_problem = Problem.from_yaml(args.yaml_file_name)
+    if args.simulation_file_name:
+        simulations_df = get_simulation_df(args.simulation_file_name)
+    else:
+        simulations_df = None
```
Comment (dilpath) [inline petab/visualize/cli.py:39]:
```suggestion
    simulations_df = None
    if args.simulation_file_name:
        simulations_df = get_simulation_df(args.simulation_file_name)
```

File: petab/visualize/cli.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    if args.visualization_file_name:
+        petab_problem.visualization_df = get_visualization_df(
+            args.visualization_file_name)
+
+    if args.style_file_name:
+        plt.style.use(args.style_file_name)
+
+    Path(args.output_directory).mkdir(exist_ok=True)
```
Comment (dilpath) [inline petab/visualize/cli.py:48]: Perhaps?
```suggestion
    Path(args.output_directory).mkdir(exist_ok=True, parents=True)
```

**Review by dilpath — APPROVED** [review]
:+1:

Could add a cell to a notebook to show how the same plot can be generated with either Python code or this CLI

File: petab/visualize/cli.py (hunk 7 above)
Comment (dweindl) [reply petab/visualize/cli.py:48]: Thx, yes

**Comment by dweindl** [comment]
> Could add a cell to a notebook to show how the same plot can be generated with either Python code or this CLI

Agreed. Will keep that for after #89 (notebooks are currently failing).

File: petab/visualize/cli.py (hunk 3 above)
Comment (dweindl) [reply petab/visualize/cli.py:29]: e.g., `--style seaborn-colorblind`

**Pull request merged**